docs/howto/advanced/priorities.md

```diff
@@ -0,0 +1,29 @@
+# Provide a job priority
+
+We can assign an optional priority to a job. Jobs with higher priority will be
+preferred by a worker. Priority is represented as an integer, where a higher
+number indicates a higher priority. If no priority is specified, it defaults
+to 0.
+
```

Comment:
I think it might be useful to state the following even it may seem a bit obvious if we give it some thought (feel free to rephrase as needed):

> Priority is used as way to order _available_ jobs. If a procrastinate worker requests a task, and the 2 scheduled tasks are a high-priority task that is blocked by a lock, and a low-priority that is available, the worker will take the low-priority task. Procrastinate will never wait for a high-priority task to become available if there are lower-priority tasks already available.

Reply:
I am a bit confused by the terms `job` and `task`. You mean ...

>  Priority is used as way to order available jobs. If a procrastinate worker requests a **_job_**, and the 2 scheduled **_jobs_** are a high-priority **_job_** that is blocked by a lock, and a low-priority that is available, the worker will take the low-priority **_job_**. Procrastinate will never wait for a high-priority **_job_** to become available if there are lower-priority **_jobs_** already available.

Or do you use the word job and task synonymously? I see jobs more as instances of a task and tasks more as a template or the code that gets executed behind it. But maybe I am pedantic 😉

Reply:
You're definitely not pedantic, it's confusing and it's my fault. The 2 terms are  explained [here](https://procrastinate.readthedocs.io/en/stable/glossary.html#). When we started the project, we needed to decide what would be the word for the function that can be deferred, and what would be the word for a function deferred with a specific set of arguments, options, etc. We kinda arbitrarily settled for "task" and "job". And the fact that I, after having spent years on the project, frequently get those mixed up, is a continued proof that these were not the right terms. I'm afraid this ship has sailed much too far away to get this fixed.

Yes, in my paragraph above, I meant `job` everywhere, tasks (for now) are unconcerned with priorities. (That said, tasks could have a default priority, as they have a default queue, lock, etc)

Reply:
Thanks for the clarification. Naming stuff is always hard. The funny thing is that in the pipeline where I want to integrate Procrastinate, it's exactly the other way around. A job consists of several tasks 😂